pennylane/io.py

```diff
@@ -0,0 +1,246 @@
+    >>>     return qml.expval(qml.PauliZ(0))
+
+    Args:
+        quantum_circuit (qiskit.QuantumCircuit): a quantum circuit created in qiskit
+
+    Returns:
+        function: the PennyLane template created based on the QuantumCircuit
+        object
+    """
+    return load(quantum_circuit, format='qiskit')
+
+
+def from_qasm(quantum_circuit: str):
+    """Loads quantum circuits from a QASM string using the converter in the
+    PennyLane-Qiskit plugin.
+
+    **Example:**
+
+    .. code-block:: python
+
+        >>> hadamard_qasm = 'OPENQASM 2.0;' \\
+        ...                 'include "qelib1.inc";' \\
+        ...                 'qreg q[1];' \\
+        ...                 'h q[0];'
+        >>> my_circuit = qml.from_qasm(hadamard_qasm)
+
+    The ``my_circuit`` template can now be used within QNodes, as a
+    two-wire quantum template.
+
+    >>> @qml.qnode(dev)
+    >>> def circuit(x):
+    >>>     qml.RX(x, wires=1)
+    >>>     my_circuit(wires=(1, 0))
+    >>>     return qml.expval(qml.PauliZ(0))
+
+    Args:
+        quantum_circuit (str): a QASM string containing a valid quantum circuit
+
+    Returns:
+        function: the PennyLane template created based on the QASM string
+    """
+    return load(quantum_circuit, format='qasm')
+
+
+def from_qasm_file(qasm_filename: str):
+    """Loads quantum circuits from a QASM file using the converter in the
+    PennyLane-Qiskit plugin.
+
+    **Example:**
+
+    >>> my_circuit = qml.from_qasm("hadamard_circuit.qasm")
+
+    The ``my_circuit`` template can now be used within QNodes, as a
+    two-wire quantum template.
+
+    >>> @qml.qnode(dev)
+    >>> def circuit(x):
+    >>>     qml.RX(x, wires=1)
+    >>>     my_circuit(wires=(1, 0))
+    >>>     return qml.expval(qml.PauliZ(0))
+
+    Args:
+        qasm_filename (str): path to a QASM file containing a valid quantum circuit
+
+    Returns:
+        function: the PennyLane template created based on the QASM file
+    """
+    return load(qasm_filename, format='qasm_file')
+
+
+def from_pyquil(pyquil_program):
+    """Loads pyquil Program objects by using the converter in the
+    PennyLane-Forest plugin.
+
+    **Example:**
+
+    >>> program = pyquil.Program()
+    >>> program += pyquil.gates.H(0)
+    >>> program += pyquil.gates.CNOT(0, 1)
+    >>> my_circuit = qml.from_pyquil(program)
+
+    The ``my_circuit`` template can now be used within QNodes, as a
+    two-wire quantum template.
+
+    >>> @qml.qnode(dev)
+    >>> def circuit(x):
+    >>>     qml.RX(x, wires=1)
+    >>>     my_circuit(wires=[1, 0])
+    >>>     return qml.expval(qml.PauliZ(0))
+
+    Args:
+        pyquil_program (pyquil.Program): a program created in pyquil
+
+    Returns:
+        pennylane_forest.ProgramLoader: a pennylane_forest.ProgramLoader instance that can
+            be used like a PennyLane template and that contains additional inspection properties
+    """
+    return load(pyquil_program, format='pyquil_program')
```

Comment:
Would there be a downside to having this as follows?

Reply:
Yes, the entry point was called `pyquil_program` in the pl-forest plugin, so renaming it will just not work. I also deliberately chose `pyquil_program` over `pyquil` because I wanted to exactly specify what kind of object will be imported

Reply:
I see! One thing, however, in this case, would be that since this is a user-facing function, users will have to keep in mind this syntax. This might lead to a caveat when they are creating code that could be avoided.

Adding the extra information of converting a pyquil ``Program`` sounds pretty reasonable, however, since there is no other type of object being converted currently, would it be necessary?

Reply:
When it comes to the entry point, I believe that could be adjusted if needed

Reply:
I also think that there only being one kind of conversion that we make not to be a good argument to rename it to `load_pyquil`. I'd actually rename the `load_qiskit` one to `load_qiskit_circuit` (or whatever the object is called).

I think otherwise people might think they can load whatever pyquil object they want, which is not true.